plugins/dubbo/src/main/java/com/megaease/easeagent/plugin/dubbo/ApacheDubboCtxUtils.java

```diff
@@ -0,0 +1,199 @@
+	//dubbo get metadata interface
+	public static final String METADATA_INTERFACE = "org.apache.dubbo.metadata.MetadataService";
+
+	private static final Logger log = LoggerFactory.getLogger(ApacheDubboCtxUtils.class);
+
+	/**
+	 * init span
+	 *
+	 * @param methodInfo
+	 * @param context
+	 * @param config
+	 */
+	public static void initSpan(MethodInfo methodInfo, Context context, DubboTraceConfig config) {
+		Invoker<?> invoker = (Invoker<?>) methodInfo.getArgs()[0];
+		Invocation invocation = (Invocation) methodInfo.getArgs()[1];
+		URL requestUrl = invoker.getUrl();
+
+		if (requestUrl.getServiceInterface().equals(METADATA_INTERFACE)) {
+			return;
+		}
+
+		RpcContext rpcContext = RpcContext.getContext();
+		boolean isConsumer = requestUrl.getParameter(SIDE_KEY).equals(CONSUMER_SIDE);
+		String applicationName = requestUrl.getParameter(CommonConstants.APPLICATION_KEY);
+		if (isConsumer) {
+			String groupName = requestUrl.getParameter(CommonConstants.GROUP_KEY);
+			String version = requestUrl.getParameter(CommonConstants.VERSION_KEY);
+
+			ApacheDubboClientRequest apacheDubboClientRequest = new ApacheDubboClientRequest(invoker, invocation);
+			RequestContext requestContext = context.clientRequest(apacheDubboClientRequest);
+			Span span = requestContext.span().start();
+			span.kind(apacheDubboClientRequest.kind());
+			span.name(apacheDubboClientRequest.name());
+			span.remoteServiceName(CommonConstants.DUBBO);
+			span.remoteIpAndPort(requestUrl.getIp(), requestUrl.getPort());
+
+			String argsList = config.argsCollectEnabled() ? JsonUtil.toJson(invocation.getArguments()) : null;
+			span.tag(DubboTags.CLIENT_APPLICATION.name, applicationName);
+			span.tag(DubboTags.GROUP.name, groupName);
+			span.tag(DubboTags.SERVICE.name, requestUrl.getPath());
+			span.tag(DubboTags.METHOD.name, method(invocation));
+			span.tag(DubboTags.SERVICE_VERSION.name, version);
+			span.tag(DubboTags.ARGS.name, argsList);
+
+			context.put(CLIENT_REQUEST_CONTEXT, requestContext);
+		} else {
+			ApacheDubboServerRequest apacheDubboServerRequest = new ApacheDubboServerRequest(invoker, invocation);
+			RequestContext requestContext = context.serverReceive(apacheDubboServerRequest);
+
+			Span span = requestContext.span().start();
+			span.kind(apacheDubboServerRequest.kind());
+			span.name(apacheDubboServerRequest.name());
+			span.remoteServiceName(CommonConstants.DUBBO);
+			span.remoteIpAndPort(rpcContext.getRemoteHost(), rpcContext.getRemotePort());
+			span.tag(DubboTags.SERVER_APPLICATION.name, applicationName);
+
+			context.put(SERVICE_REQUEST_CONTEXT, requestContext);
+		}
+	}
+
+	public static void finishSpan(URL url, DubboTraceConfig config, Context context, Result result, Throwable throwable) {
+		boolean isConsumer = url.getParameter(SIDE_KEY).equals(CONSUMER_SIDE);
+		if (result instanceof AsyncRpcResult) {
+			result.whenCompleteWithContext(new DubboAsyncCallback(isConsumer, config, context, throwable));
```

Comment:
`requestContext.scope().close();` must be executed in a method on the same thread as `before`. But you are passing it to other threads for execution, which is invalid